src/models/converters/safes.rs

```diff
@@ -49,22 +58,32 @@ impl SafeInfo {
 
 pub(super) fn calculate_version_state(
     safe_version: &str,
-    min_chain_version: &Option<String>,
+    supported_master_copies: &Vec<MasterCopy>,
+    min_chain_version: String,
 ) -> ImplementationVersionState {
-    if let Some(min_chain_version) = min_chain_version {
-        let sem_ver_safe = Version::parse(safe_version);
-        let sem_ver_min = Version::parse(min_chain_version);
+    let sem_ver_safe = Version::parse(safe_version);
+    let sem_ver_min = Version::parse(&min_chain_version);
 
-        if sem_ver_min.is_err() || sem_ver_safe.is_err() {
-            return ImplementationVersionState::Unknown;
-        }
+    let supported_versions = supported_master_copies
+        .iter()
+        .filter_map(|it| Version::parse(&it.version).ok())
+        .collect::<Vec<Version>>();
+
+    if sem_ver_min.is_err() || sem_ver_safe.is_err() || supported_versions.is_empty() {
+        return ImplementationVersionState::Unknown;
+    }
+
+    let sem_ver_safe = sem_ver_safe.unwrap();
+    let sem_ver_min = sem_ver_min.unwrap();
 
-        match sem_ver_safe.unwrap().cmp(&sem_ver_min.unwrap()) {
-            Ordering::Less => ImplementationVersionState::Outdated,
-            Ordering::Equal => ImplementationVersionState::UpToDate,
-            Ordering::Greater => ImplementationVersionState::UpToDate,
+    match sem_ver_safe.cmp(&sem_ver_min) {
+        Ordering::Less => {
+            if supported_versions.contains(&sem_ver_safe) {
+                ImplementationVersionState::Outdated
+            } else {
+                ImplementationVersionState::Unknown
+            }
         }
-    } else {
-        ImplementationVersionState::Unknown
+        Ordering::Equal | Ordering::Greater => ImplementationVersionState::UpToDate,
```

Comment:
Why do we not check the supported versions in this case? This should be checked always.

I would do the following steps:
1. Load Safe info
2. Load supported master copies
3. Check that Safe master copy is part of supported master copies
4. Load recommended version
5. Map state

Reply:
I would add this to the earlier return condition in line ~72